pymeasure/adapters/telnet.py

```diff
@@ -0,0 +1,114 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import telnetlib
+import time
+from warnings import warn
+
+from .adapter import Adapter
+
+
+class TelnetAdapter(Adapter):
+    """ Adapter class for using the Python telnetlib package to allow
+    communication to instruments
+
+    .. deprecated:: 0.11.2
+        Preparation for telnetlib deprecation. 
+        TelnetAdapter is deprecated, use VISAAdapter instead.
```

Comment:
It would be great to tell users _how_ to do that -- what do they have to do to a VISAAdapter so that it behaves like a TelnetAdapter?

Reply:
Thanks, this seems appropriate.

```suggestion
        telnetlib is deprecated in Python 3.11 and will be removed in Python 3.13.
        As a result, TelnetAdapter is deprecated, use VISAAdapter instead.
        The VISAAdapter supports TCPIP socket connections. When using the VISAAdapter,
        the `resource_name` argument should be `TCPIP[board]::<host>::<port>::SOCKET`.
        see here, <https://pyvisa.readthedocs.io/en/latest/introduction/names.html>
```

Reply:
Ah, that is all? There's no additional configuration flags or somesuch necessary? That's Great!

Reply:
Yeah, the TelnetAdapter previously implemented everything that the telnetlib was missing from Pymeasures aspect. The python telnet client was always very simple (`class telnetlib.Telnet(host=None, port=0[, timeout])`). All the methods of Telnet were simple reads and writes. VISAAdapter handles these. The TelnetAdapter implemented write_termination and read_termination, pyvisa handles this. It's fairly complete.